src/app/store/state/product.state.ts

```diff
@@ -0,0 +1,15 @@
+import {Product} from "../../model/product";
+
+export interface ProductState {
+  products: Product[];
+  selectedProduct : Product | undefined;
+  error: string;
+  status: 'pending' | 'loading' | 'error' | 'success';
```

Comment:
This should be a type, since you're reusing it. Otherwise, in a lot of projects, it's typically a boolean value called loading.